docs/developer/set-goals.md

```diff
@@ -79,11 +79,47 @@ The DSL includes support for logical operations. For example, this will build al
 whenPushSatisfies(IsMaven, not(IsKotlin)).setGoals(BuildGoals),
 ```
 
-## PushRule
-
 Each argument to [`sdm.withPushRules`](https://atomist.github.io/sdm/interfaces/_lib_api_machine_softwaredeliverymachine_.softwaredeliverymachine.html#withpushrules)
 is a PushRule, contributing goals on a commit if a condition is met. That condition is a PushTest.
 
+### Advanced Push Rules
```

Comment:
I think linking to a section specifically on push rules would make sense here, but we really only talk about [push tests](https://docs.atomist.com/developer/push-test/). Should that article be made more generic to talk about how push tests can be a component of a push rule?

Reply:
Push Tests also apply to code transforms, and sometimes other things, so it makes sense for them to have their own section